tests/unit/parcelset/put_parcels_with_coalescing.cpp

```diff
@@ -0,0 +1,241 @@
+    for (int i = 0; i != numparcels_default; ++i)
+    {
+        hpx::lcos::promise<hpx::id_type> p_cont;
+
+        if (std::rand() % 2)
+        {
+            parcels.push_back(
+                generate_parcel<test1_action>(id, p_cont.get_id(), data)
+            );
+        }
+        else
+        {
+            hpx::lcos::local::promise<double> p_arg;
+
+            parcels.push_back(
+                generate_parcel<test2_action>(id, p_cont.get_id(),
+                    p_arg.get_future())
+            );
+
+            args.push_back(std::move(p_arg));
+        }
+
+        results.push_back(p_cont.get_future());
+    }
+
+    // send parcels
+    hpx::get_runtime().get_parcel_handler().put_parcels(std::move(parcels));
+
+    // now make the futures ready
+    for (hpx::lcos::local::promise<double>& arg : args)
+    {
+        arg.set_value(42.0);
+    }
+
+    // verify all messages got actually sent to the correct locality
+    hpx::wait_all(results);
+
+    for (hpx::future<hpx::id_type>& f : results)
+    {
+        HPX_TEST(f.get() == id);
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+void print_counters(char const* name)
+{
+    using namespace hpx::performance_counters;
+
+    std::vector<performance_counter> counters = discover_counters(name);
+
+    for (performance_counter const& c : counters)
+    {
+        counter_value value = c.get_counter_value_sync();
+        hpx::cout
+            << "counter: " << c.get_name_sync()
+            << ", value: " << value.get_value<double>()
+            << std::endl;
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main(boost::program_options::variables_map& vm)
+{
+    unsigned int seed = (unsigned int)std::time(0);
+    if (vm.count("seed"))
+        seed = vm["seed"].as<unsigned int>();
+
+    std::cout << "using seed: " << seed << std::endl;
+    std::srand(seed);
+
+    for (hpx::id_type const& id : hpx::find_remote_localities())
+    {
+        test_plain_argument(id);
+        test_future_argument(id);
+        test_mixed_arguments(id);
+    }
+
+    // compare number of parcels with number of messages generated
+    print_counters("/parcels/count/*/sent");
+    print_counters("/messages/count/*/sent");
+
+    return hpx::finalize();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int main(int argc, char* argv[])
+{
+    // add command line option which controls the random number generator seed
+    using namespace boost::program_options;
+    options_description desc_commandline(
+        "Usage: " HPX_APPLICATION_STRING " [options]");
+
+    desc_commandline.add_options()
+        ("seed,s", value<unsigned int>(),
+        "the random number generator seed to use for this run")
+        ;
+
+    // explicitly enable message handlers (parcel coalescing)
+    std::vector<std::string> cfg;
+    cfg.push_back("hpx.parcel.message_handlers=1");
```

Comment:
This enables it at runtime. The test itself is compiled only if the module is being enabled, though (see [here](https://github.com/STEllAR-GROUP/hpx/pull/1942/files#diff-4cd492c630d6c13e035d83f762c9ccacR17)).